src/modules/rewards/services/arb-rebate-service.ts

```diff
@@ -268,18 +267,25 @@ export class ArbRebateService {
       };
     }
 
+    const inputTokenPrice = deposit.price.usd;
     const historicRewardTokenPrice = await this.marketPriceService.getCachedHistoricMarketPrice(
       DateTime.fromJSDate(deposit.depositDate).minus({ days: 1 }).toJSDate(),
       rewardToken.symbol.toLowerCase(),
     );
+
+    const inputAmountUsd = new BigNumber(deposit.amount)
+      .multipliedBy(inputTokenPrice)
+      .dividedBy(new BigNumber(10).pow(deposit.token.decimals));
+    const maxRewardsUsd = inputAmountUsd.multipliedBy(REWARDS_PERCENTAGE_LIMIT);
     const rewardsUsd = new BigNumber(bridgeFeeUsd).multipliedBy(ARB_REBATE_RATE).toFixed();
+    const cappedRewardsUsd = BigNumber.min(maxRewardsUsd, rewardsUsd).toFixed();
     const rewardsAmount = ethers.utils.parseEther(
-      new BigNumber(rewardsUsd).dividedBy(historicRewardTokenPrice.usd).toFixed(18),
+      new BigNumber(cappedRewardsUsd).dividedBy(historicRewardTokenPrice.usd).toFixed(18),
     );
 
     return {
       rewardToken,
-      rewardsUsd,
+      rewardsUsd: cappedRewardsUsd,
```

Comment:
This looks to cap the reward amount in USD. Should we also want to cap the reward amount in terms of the token (for `rewardsAmount` below)

Reply:
Since `rewardsAmount` is calculated using `rewardsUsd` which is capped, I'd say there's no need to cap it in terms of the reward token. But lmk if I'm missing something here.